### PR TITLE
Agent housekeeping

### PR DESCRIPTION
**Pull request by Razz4780**
Some minor improvements to agent code:
1. `TcpStealerApi` no longer need to hold a shared channel permit to close session with the background task
2. Removed last uses `IndexAllocator`
3. Added some more logs to outgoing task
4. Improved agent errors

### CODE REVIEW

File: mirrord/agent/src/outgoing.rs [hunk 1]
```diff
@@ -89,7 +90,7 @@ impl TcpOutgoingApi {
     }
 
     /// Receives a [`DaemonTcpOutgoing`] message from the background task.
-    pub(crate) async fn daemon_message(&mut self) -> Result<DaemonTcpOutgoing> {
+    pub(crate) async fn recv_from_task(&mut self) -> Result<DaemonTcpOutgoing> {
```
Comment (meowjesty) [inline mirrord/agent/src/outgoing.rs:93]:
```suggestion
    #[tracing::instrument(level = Level::TRACE, skip(self), err)]
    pub(crate) async fn recv_from_task(&mut self) -> Result<DaemonTcpOutgoing> {
```

Add tracing here?

File: mirrord/agent/src/outgoing.rs [hunk 2]
```diff
@@ -148,45 +149,51 @@ impl TcpOutgoingTask {
     }
 
     /// Runs this task as long as the channels connecting it with [`TcpOutgoingApi`] are open.
+    /// This routine never fails and returns [`Result`] only due to [`WatchedTask`] constraints.
+    #[tracing::instrument(level = Level::TRACE, skip(self))]
```
Comment (meowjesty) [inline mirrord/agent/src/outgoing.rs:153]:
```suggestion
    #[tracing::instrument(level = Level::TRACE, skip(self), err)]
```

We're going to see the error elsewhere, but maybe add it in the tracing here just for consistency?

File: mirrord/agent/src/runtime/error.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            error: error.to_string(),
+        }
+    }
+
+    pub(crate) fn unknown_runtime<N: ToString>(name: N) -> Self {
+        Self::UnknownRuntimeName(name.to_string())
+    }
+}
+
+pub(crate) type Result<T, E = ContainerRuntimeError> = core::result::Result<T, E>;
```
Comment (meowjesty) [inline mirrord/agent/src/runtime/error.rs:39]: This pattern kinda always annoys me, when the alias is named just `Result`, and I want to check something out by writing a quick function that returns a `Result<(), SomeOtherError>`.

Could you rename it to `ContainerResult` (or something else you find appropriate)?

**Review by meowjesty — APPROVED** [review]
Approving, as I only have minor stuff to comment on.

File: mirrord/agent/src/outgoing.rs (hunk 2 above)
Comment (Razz4780) [reply mirrord/agent/src/outgoing.rs:153]: There is no error, this task never fails :sweat_smile:

Comment (meowjesty) [reply mirrord/agent/src/outgoing.rs:153]: Yup, saw it after the comment ...

**Pull request merged**